File: e2e-pytest-playwright/pages/flight_page.py
from pages.abstract_page import AbstractPage

class FlightPage(AbstractPage):
  
  def select_oneway_trip(self):
    self.page.locator("xpath=//input[@name='tripType' and @value='oneway']").click()

  def select_return_trip(self):
    self.page.locator("xpath=//input[@name='tripType' and @value='return']").click()


  def select_depart_from(self, city):
    self.page.select_option("select[name='fromPort']", label=city)

  def select_arrive_at(self, city):
    self.page.select_option("select[name='toPort']", label=city)

  def select_departure_day(self, day):
    self.page.select_option("#departDay", label=day)

  def select_departure_month(self, month_year):
    self.page.select_option("#departMonth", label=month_year)

  def select_return_day(self, day):
    self.page.select_option("#returnDay", label=day)

  def select_return_month(self, month_year):
    self.page.select_option("#returnMonth", label=month_year)
    
  def click_continue(self):
    self.page.locator("xpath=//input[@value='Continue']").click()

File: e2e-playwright-test-refactored/pages/flight_page.js
const AbstractPage = require('./abstract_page');

class FlightPage extends AbstractPage {

  constructor(page) {
    super(page);
  }

  async selectTripType(trip_type) {
    const radios = await this.page.$$("input[name=tripType]");
    if (trip_type == "oneway") {
      await radios[1].check();
    } else  {
        await radios[0].check();
    }
  }

  async selectDepartFrom(city) {
    await this.page.selectOption("select[name='fromPort']", city);
  }

  async selectArriveAt(city) {
    await this.page.selectOption("select[name='toPort']", city);
  }

  async selectDepartDay(day) {
    await this.page.selectOption("select[name='departDay']", day);
  }

  async selectDepartMonth(month_year) {
     await this.page.selectOption("#departMonth", month_year);
  }

  async selectReturnDay(day) {
    await this.page.selectOption("select[name='returnDay']", day);

  }

  async selectReturnMonth(month_year) {
    await this.page.selectOption("#returnMonth", month_year);
  }

  async clickContinue() {
    await this.page.click("input:has-text('Continue')");
  }
};

module.exports = FlightPage;
